cpp/tests/groupby/collect_set_test.cpp

```diff
@@ -34,6 +34,7 @@ namespace test {
 #define COLLECT_SET cudf::make_collect_set_aggregation()
 #define COLLECT_SET_NULL_UNEQUAL \
   cudf::make_collect_set_aggregation(null_policy::INCLUDE, null_equality::UNEQUAL)
+#define COLLECT_SET_NULL_EXCLUDE cudf::make_collect_set_aggregation(null_policy::EXCLUDE)
```

Comment:
@sperlingxx: I understand that you're following the convention established in this test file already. But this should have ideally been defined as a function or constant, not as a pre-processor macro.

This line is particularly egregious:
```c++
#define COLLECT_SET cudf::make_collect_set_aggregation()
```
@ttnghia and @davidwendt will correct me if I'm wrong: The macro defined above would make it impossible to use the enum `aggregation::COLLECT_SET` in the body of the test. The pre-processor will substitute it, and present compile failures.

It would be better to switch these to functions/constants, and avoid macros for this sort of thing.

Reply:
I've replaced all macros for collect_set.

Reply:
This works, of course. But @ttnghia might not appreciate the added verbosity. You could, instead, use a function:
```c++
auto COLLECT_SET() { return cudf::make_collect_set_aggregation(); }
auto COLLECT_SET_NULL_UNEQUAL() { return cudf::make_collect_set_aggregation(null_policy::INCLUDE, null_equality::UNEQUAL); }
// ...
test_single_agg(keys, vals, keys_expected, vals_expected, COLLECT_SET());
```
You should now be able to skip the `std::move()`, and the `agg` definition. At the cost of needing parentheses (`()`), we have a safer bit of code.

Reply:
That's approach is great. I would suggest to use lower case `collect_set()` and `collect_set_null_unequal()`.

Reply:
Great suggestion! I've applied it :)